include/stronk/stronk.h

```diff
@@ -6,62 +6,78 @@
 #include <utility>
 
 #include <stronk/utilities/equality.h>
+#include <stronk/utilities/macros.h>
 #include <stronk/utilities/strings.h>
 
 namespace twig
 {
+
+template<typename T>
+concept should_be_copy_constructed = std::is_trivially_copyable_v<T> && sizeof(T) <= sizeof(T*);
+
 template<typename Tag, typename T, template<typename> typename... Skills>
 struct stronk : public Skills<Tag>...
 {
-  private:
-    T _v;
-
-  public:
     using underlying_type = T;
+    // we need this value to be publicly available for stronk types to be usable as non-type template parameters.
```

Comment:
Is it possible to explain why this is required to use non-type template parameters? Or does that become too complicated?